src/CollapsibleItem.js

```diff
@@ -31,9 +31,15 @@ class CollapsibleItem extends Component {
 
   render() {
     const { name, label, children } = this.props;
+    const rest = Object.keys(this.props)
+      .filter(prop => !CollapsibleItem.propNames.includes(prop))
+      .reduce((acc, prop) => {
+        acc[prop] = this.props[prop];
+        return acc;
+      }, {});
```

Comment:
I guess you are doing this instead of
`const { collapsed, name, label, children, toggle, ...rest } = this.props;`

because eslint complains about unused variables? Would it be better to disable the specifc rule for this line?

Reply:
I do it not because of linter, but because of react. Without it there will be warnings like:
`Invalid value for prop `toggle` on <div> tag`
more info: https://reactjs.org/blog/2017/09/08/dom-attributes-in-react-16.html#changes-in-detail

Reply:
Wait, I don't get it. Why would it complain?

If you used `<Item {...this.props}>` then yes. But isn't the line I proposed doing exactly the same as the current code? Destructuring into `rest` only the props that are not [collapsed, name, label, children, toggle].

I'd appreciate some input on what I'm missing here.

Reply:
ahhh. Sorry, misread it yesterday. Why I use filter instead of destruction:

1. In render method we don't use some of props, so they will be unused vars. Maybe I can trick linter to ignore them, but it is confusing for a reader.
2. The main point why I do it here like this: don't maintain the list of props twice. It's way too easy to add a prop in `propTypes` and forget to add in destruction. Especially if it's an unused variable.

Reply:
Actually eslint does not complain about unused vars, it must figure out that they are needed to filter the destructuring. And if extra props end up in `rest`, react will complain.

Anyway, I get your point, I'll leave it up to you to decide :+1: